components/script/dom/cssrule.rs

```diff
@@ -198,3 +264,31 @@ pub(crate) trait SpecificCSSRule {
         // most CSSRules do nothing here
     }
 }
+
+pub(crate) struct RulesModificationScope<'stylesheet> {
```

Comment:
I'm not really seeing the point of this, wouldn't it be simpler to manually call `stylesheet.notify_invalidations()` after modifications?